encode.go

```diff
@@ -320,6 +320,15 @@ func (e *encoder) stringv(tag string, in reflect.Value) {
 		// there's no need to quote it.
 		rtag, _ := resolve("", s)
 		canUsePlain = rtag == strTag && !isBase60Float(s)
+
+		// Maintain some backward compatibility with YAML 1.2
+		if rtag == strTag {
```

Comment:
This would be slightly nicer if we had it inside something like _isOldBool_ and put the test next to the isBase60Float above, as that method is the exactly equivalent except for base 60 floats (!!!, and you thought the on/off situation was bad).

Checking for isBase60Float usage should find us the exact places we need to care about. Well, either that or we have two bugs instead of one.. ;-)

Reply:
Okay, have moved the check out and added the test to the above line. it does read more clearly.